mkosi/distributions/gentoo.py

```diff
@@ -211,43 +170,50 @@ def fetch_fix_stage3(self) -> None:
             else:
                 die("profile names changed upstream?")
 
+        stage3_tar_path = state.cache_dir / stage3_tar
+        cls.stage3_cache = stage3_tar_path.with_name(stage3_tar.name).with_suffix(".tmp")
+
+        user_config_path = cls.stage3_cache / cls.USER_CONFIG_PATH
+
+        emerge_vars = {
+            "FEATURES": " ".join(cls.portage_features),
+            "USE": " ".join(cls.portage_use_flags),
+        }
+
+        for d in ("binpkgs", "distfiles", "repos"):
+            state.cache_dir.joinpath(d).mkdir(exist_ok=True)
+
         stage3_url_path = urllib.parse.urljoin(
-            self.portage_cfg["GENTOO_MIRRORS"],
-            f"releases/{self.arch}/autobuilds/{stage3_tar}",
+            cls.gentoo_mirrors,
+            f"releases/{arch}/autobuilds/{stage3_tar}",
         )
-        stage3_tar_path = self.state.cache_dir / stage3_tar
-        stage3_tmp_extract = stage3_tar_path.with_name(stage3_tar.name + ".tmp")
         if not stage3_tar_path.is_file():
             log_step(f"Fetching {stage3_url_path}")
             stage3_tar_path.parent.mkdir(parents=True, exist_ok=True)
             urllib.request.urlretrieve(stage3_url_path, stage3_tar_path)
 
-        stage3_tmp_extract.mkdir(parents=True, exist_ok=True)
+        cls.stage3_cache.mkdir(parents=True, exist_ok=True)
 
-        with flock(stage3_tmp_extract):
-            if not stage3_tmp_extract.joinpath(".cache_isclean").exists():
-                log_step(f"Extracting {stage3_tar.name} to {stage3_tmp_extract}")
+        with flock(cls.stage3_cache):
```

Comment:
This shouldn't be necessary. If this is required, we should lock the cache directory itself for all distros, not just gentoo

Reply:
Yeah sure we can keep the cache_isclean but let's drop the flock()

Reply:
dropped the `flock`